pkg/dockermanager/dockermanager.go

```diff
@@ -123,7 +126,24 @@ func (dm *Manager) Manage(
 		if m, ok := managers[container]; ok {
 			m.cancel()
 			delete(managers, container)
-			<-m.done
+
+			timeout := time.NewTimer(workerShutdownTimeout)
+			defer timeout.Stop()
+
+			ticker := time.NewTicker(workerShutdownTick)
+			defer ticker.Stop()
+
+			for {
+				select {
+				case <-m.done:
+					return
+				case <-timeout.C:
+					dm.S().Panicw("timed out waiting for container worker to stop", "container", container)
```

Comment:
I'm not sure why we would ever panic the sidecar on a problem with a container.

If `docker` (or k8s api) goes berserk and doesn't remove a container, we should just flood `sidecar` with errors, but not panic it. One container should not be bringing the sidecar on a given host down.

Reply:
To put more context here, if we run test plans overnight and there is a problem with Docker for some reason, I don't want to see `sidecar` having restarted 200 times, because we somehow managed to break another part of the system - I'd like to see 200 errors and have to figure out what has gone wrong.

With respect to the hanging go-routines - just close them after a timeout for containers that are not responding if this ever happens, and log an error.

Reply:
This is a problem in the sidecar, not containers. We'll only hit this if the sidecar is stuck doing something and isn't returning from the worker. (see https://github.com/ipfs/testground/pull/456 for a case where that can happen).

I can't close hanging goroutines. Go doesn't let me do that.

Basically, if the sidecar is stuck, there's a bug somewhere in the sidecar and we need to do _something_. Crashing is about the best thing we can do.

Reply:
OK, then I misunderstood what this is about!